File: doclens/web_v2/frontend/src/api/vision.ts
/** /api/vision/* 客户端：默认提示词预填 + 图像重新解析。 */
import { ApiError, request } from "./client";

export interface PromptResponse {
  prompt: string;
}

export interface ReparseResponse {
  path: string;
  markdown: string;
}

export class ReparseError extends Error {
  constructor(public code: string, message: string, public status: number) {
    super(message);
    this.name = "ReparseError";
  }
}

/** 拉取默认 VISION_PROMPT，供对话框预填（单一真相源在后端）。 */
export async function getVisionPrompt(): Promise<string> {
  const r = await request<PromptResponse>("/api/vision/prompt");
  return r.prompt;
}

/** 用自定义提示词重新解析一张图像，返回新 markdown。 */
export async function reparseImage(
  path: string,
  prompt: string,
): Promise<ReparseResponse> {
  try {
    return await request<ReparseResponse>("/api/vision/reparse", {
      method: "POST",
      json: { path, prompt },
    });
  } catch (e) {
    if (e instanceof ApiError) {
      throw new ReparseError(e.code, e.message, e.status);
    }
    throw e;
  }
}

/** 手动备注：用户 markdown 直接覆盖 AI 解读（不调视觉模型，持久保留）。 */
export async function saveManualNote(
  path: string,
  markdown: string,
): Promise<ReparseResponse> {
  try {
    return await request<ReparseResponse>("/api/vision/note", {
      method: "POST",
      json: { path, markdown },
    });
  } catch (e) {
    if (e instanceof ApiError) {
      throw new ReparseError(e.code, e.message, e.status);
    }
    throw e;
  }
}

File: doclens/web_v2/models/vision.py
"""Pydantic 模型：/api/vision/* 端点的请求/响应。"""
from pydantic import BaseModel


class PromptResponse(BaseModel):
    """GET /api/vision/prompt 响应：返回默认 VISION_PROMPT 供前端对话框预填。"""
    prompt: str


class ReparseRequest(BaseModel):
    """POST /api/vision/reparse 请求体。

    ``path`` 相对 workdir 的图像路径；``prompt`` 用户编辑后的自定义提示词
    （覆盖默认 VISION_PROMPT 仅对本次解析生效）。
    """
    path: str
    prompt: str


class ReparseResponse(BaseModel):
    """POST /api/vision/reparse 响应。"""
    path: str
    markdown: str


class NoteRequest(BaseModel):
    """POST /api/vision/note 请求体：用户手写备注直接覆盖 AI 解读（不调视觉模型）。"""
    path: str
    markdown: str
